File: src/main/java/gargoyle/rpycg/util/Check.java
package gargoyle.rpycg.util;

import java.util.regex.Pattern;

public final class Check {
    private static final Pattern PATTERN_ID = Pattern.compile("^[_a-zA-Z](?:[._a-zA-Z0-9]*[_a-zA-Z0-9]+)?$");
    private static final Pattern PATTERN_TEXT = Pattern.compile("^[^\n;]+$");

    private Check() {
        throw new IllegalStateException(Check.class.getName());
    }

    public static boolean isFloat(String val) {
        try {
            Float.parseFloat(val);
            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }

    public static boolean isIdentifier(CharSequence id) {
        return PATTERN_ID.matcher(id).matches();
    }

    public static boolean isInteger(String val) {
        try {
            Integer.parseInt(val);
            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }

    public static boolean isText(CharSequence text) {
        return PATTERN_TEXT.matcher(text).matches();
    }
}

File: app/src/main/kotlin/gargoyle/rpycg/util/Check.kt
package gargoyle.rpycg.util

object Check {
    private val PATTERN_ID = "^[_a-zA-Z](?:[._a-zA-Z0-9]*[_a-zA-Z0-9]+)?$".toRegex()
    private val PATTERN_TEXT = "^[^\n;]+$".toRegex()

    fun isFloat(value: String): Boolean =
        null != value.toFloatOrNull()

    fun isIdentifier(id: CharSequence): Boolean =
        PATTERN_ID.matches(id)

    fun isInteger(value: String): Boolean =
        null != value.toIntOrNull()

    fun isText(text: CharSequence): Boolean =
        PATTERN_TEXT.matches(text)

}
